packages/eslint-plugin/tests/rules/no-unnecessary-condition.test.ts

```diff
@@ -253,6 +253,26 @@ function test<T>(a: T) {
   const t16 = undefined !== a;
 }
     `,
+    `
+function foo<T extends object>(arg: T, key: keyof T): void {
```

Comment:
[Testing] Per #7319, we generally prefer small & single-purpose unit tests. I think this one and the later one with three variables is testing two areas:

* That `T extends object` -> `key: keyof T` -> `arg[key]` work in general
* Different kinds of nullish checking against that `arg[key]`, specifically

I think the fact that different kinds of nullish checking work is covered nicely in other cases. So this one can probably reduce down to just one check. What do you think?

Reply:
Hi @JoshuaKGoldberg 
> Per https://github.com/typescript-eslint/typescript-eslint/pull/7319, we generally prefer small & single-purpose unit tests.

Agree!. 
I want to make sure I understand it well before I make any changes.
Maybe should I change it to something like the suggestions below?
- https://github.com/typescript-eslint/typescript-eslint/pull/8235/files#r1450814114
- https://github.com/typescript-eslint/typescript-eslint/pull/8235/files#r1450814477

Reply:
Yup, that looks right to me!